tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -271,16 +280,21 @@ GetFriendEntries(const std::vector<std::pair<std::string, std::string>> &friendN
 }
 
 ////////////////////////////////////////////////////////////////////////
-/// Return the full path of the tree
-static std::string GetTreeFullPath(const TTree &tree)
+/// Return the full path of the TTree or the trees in the TChain
+static std::vector<std::string> GetTreeFullPaths(const TTree &tree)
 {
-   // Case 1: this is a TChain: we get the name out of the first TChainElement
-   if (0 == strcmp("TChain", tree.ClassName())) {
-      auto &chain = dynamic_cast<const TChain &>(tree);
+   // Case 1: this is a TChain: for the file it contains, GetName corresponds to the name of the tree
+   if (tree.IsA() == TChain::Class()) {
+      auto &chain = static_cast<const TChain &>(tree);
       auto files = chain.GetListOfFiles();
-      if (files && 0 != files->GetEntries()) {
-         return files->At(0)->GetName();
+      if (!files || files->GetEntries() == 0) {
+         throw std::runtime_error("TTreeProcessorMT: input TChain does not contain any file");
       }
+      std::vector<std::string> treeNames;
+      for (TObject *f : *files)
+         treeNames.emplace_back(f->GetName());
```

Comment:
So when you do `GetName` on a TFile that contains exactly one tree, it returns the name of the tree?

Reply:
It's not a `TFile` thing, it's a "file-coming-from-TChain" thing. Line 286 was supposed to clarify this.

Reply:
So it is the `TChain` that sets the names of its files to be the names of the trees each file contains (which form the chain)?

Reply:
Ok thanks. I would do a slight rewording of the comment: `// Case 1: this is a TChain: for each file it contains, GetName returns the name of the tree in that file`.

Reply:
fixed